python/tank/templatekey.py

```diff
@@ -292,31 +304,164 @@ def __init__(self,
                                          abstract=abstract,
                                          length=length)
 
-        if not(format_spec is None or isinstance(format_spec, basestring)):
-            msg = "Format_spec for TemplateKey %s is not of type string: %s"
+    def _init_format_spec(self, name, format_spec):
+        """
+        Asserts that the format_spec parameter is a valid value.
+
+        :param name: Name of this template key.
+        :param format_spec: Parameter to be validated.
+
+        :raises TankError: Raised when the parameter is not a string that maching a %d format
+                           option.
+        """
+        # No format spec means no formatting options.
+        if format_spec is None:
+            self._padding_char = None
+            self._minimum_width = None
+            self.format_spec = None
+            return
+
+        if isinstance(format_spec, basestring) is False:
+            msg = "format_spec for IntegerKey %s is not of type string: %s"
             raise TankError(msg % (name, str(format_spec)))
 
+        if len(format_spec) == 0:
+            raise TankError("format_spec can't be empty.")
+
+        matches = self._FORMAT_SPEC_RE.match(format_spec)
```

Comment:
could this be something like

```python
if not(format_spec.isdigit()):
  # format spec can only contain 0-9
  raise TankError("no no no!")

elif format_spec == "0":
  # single zero is not correct
  raise TankError("no no no!")

elif not format_spec.startswith("0"):
  # positive integer, e.g. "12345", "12" or "3"
  self._zero_padded = False
  self._digit_width = int(format_spec)

elif format_spec.startswith("0") and not format_spec[1:].startswith("0"):
  # first digit is a zero and second digit isn't
  self._zero_padded = False
  self._digit_width = int(format_spec[1:])

else:
  # not valid!
  raise TankError("invalid!")
```

Just wondering if it makes sense to avoid some of that complex regex?